packages/libs/react-ui/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  checkpointContainerStyle,
+  circleColorVariant,
+  circleStyle,
+  lineColorVariant,
+  lineContainerStyle,
+  lineStyle,
+  progressBarStyle,
+  textContainerStyle,
+} from './ProgressBar.css';
+
+import classNames from 'classnames';
+import React, { FC } from 'react';
+
+export interface IProgressBarProps {
+  checkpoints: ICheckpoint[];
+}
+
+export interface ICheckpoint {
+  title: string;
+  status: 'complete' | 'pending' | 'incomplete';
+}
+
+export const ProgressBar: FC<IProgressBarProps> = ({ checkpoints }) => {
+  return (
+    <div
+      className={progressBarStyle}
+      data-testid="kda-progress-bar"
+      key={'progress-bar'}
+    >
+      {checkpoints.map((checkpoint, index) => {
+        return (
+          <>
+            {index !== 0 ? (
+              <div className={lineContainerStyle}>
+                <div
+                  className={classNames(
+                    lineStyle,
+                    lineColorVariant[checkpoint.status],
+                  )}
+                />
+                <div
+                  className={classNames(
+                    lineStyle,
+                    lineColorVariant[checkpoint.status],
+                  )}
+                />
+                <div
+                  className={classNames(
+                    lineStyle,
+                    lineColorVariant[checkpoint.status],
+                  )}
+                />
+              </div>
+            ) : null}
+            <div
+              className={checkpointContainerStyle}
+              key={index}
+              data-testid={`kda-checkpoint-container-${index}`}
+            >
+              <div
+                className={classNames(
+                  circleStyle,
+                  circleColorVariant[checkpoint.status],
+                )}
+              />
+              <div className={textContainerStyle}>{checkpoint.title}</div>
```

Comment:
I see that the text has different styles depending on the status, but I don't think they are being applied to this component
<img width="604" alt="image" src="https://github.com/kadena-community/kadena.js/assets/9022549/79045e44-176a-4b46-bfaf-d0460134a8b5">